core/src/main/scala/tech/beshu/ror/utils/CaseMappingEquality.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.utils
+
+import cats._
+import cats.implicits._
+import tech.beshu.ror.accesscontrol.domain.{Action, IndexName, RepositoryName, SnapshotName}
+
+import language.implicitConversions
+
+trait CaseMappingEquality[A] {
```

Comment:
IMO we don't need it.

`sealed trait UsernameCaseMapping` and `def eqUserId(implicit caseCompatisonStrategy: UsernameCaseMapping): Eq[User.Id]` are enough

Reply:
`Eq` doesn't know about string patterns. To test pattern against `A` `mapCases(show(A))`, and pattern is mapped too `mapCases(pattern)`.